### PR TITLE
ISPN-13479 NPE in RemoteStore when not properly initialized

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.redhat.com/browse/ISPN-13479

### CODE REVIEW

File: persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java [hunk 1]
```diff
@@ -207,8 +208,12 @@
 
    @Override
    public CompletionStage<Boolean> isAvailable() {
-      return remoteCache.ping()
-            .handle((v, t) -> t == null && v.isSuccess());
+      if (remoteCache != null) {
+         return remoteCache.ping()
+               .handle((v, t) -> t == null && v.isSuccess());
+      } else {
+         return CompletableFuture.completedFuture(false);
```
Comment (pruivo) [inline persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java:215]:
```suggestion
         return CompletableFutures.completedFalse();
```

File: persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java [hunk 2]
```diff
@@ -377,7 +382,11 @@ private Object getValue(MarshallableEntry entry) {
 
    @Override
    public CompletionStage<Void> clear() {
-      return remoteCache.clearAsync();
+      if (remoteCache != null) {
+         return remoteCache.clearAsync();
+      } else {
+         return CompletableFuture.completedFuture(null);
```
Comment (pruivo) [inline persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java:388]:
```suggestion
         return CompletableFutures.completedNull();
```

**Review by pruivo — CHANGES_REQUESTED** [review]

**Comment by diegolovison** [comment]
Done

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
merged! thanks @diegolovison !

**Pull request merged**